Add Check Support for Java 21 Record Pattern Syntax : AbbreviationAsWordInName

child of https://github.com/checkstyle/checkstyle/issues/14890
I have read check documentation: https://checkstyle.org/checks/naming/abbreviationaswordinname.html#AbbreviationAsWordInName
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words



```
PS D:\CS\test> cat src/Test2.java                                                
record ColoredPoint(int p, int x, String c) { }
record Rectangle(ColoredPoint upperLeft, ColoredPoint lowerRight) { }
public class Test2 {
    void test(Object obj) {
        if (!(obj instanceof Rectangle(ColoredPoint POINT,ColoredPoint ppPOINT))) {  // 2 violations
        }
    }

}
PS D:\CS\test> cat config.xml                                                    
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>
    <module name="TreeWalker">
        <module name="AbbreviationAsWordInName">

        </module>

    </module>
</module>
PS D:\CS\test> java  -jar checkstyle-10.17.0-all.jar -c config.xml src/Test2.java
Starting audit...
[ERROR] D:\CS\test\src\Test2.java:5:53: Abbreviation in name 'POINT' must contain no more than '4' consecutive capital letters. [AbbreviationAsWordInName]
[ERROR] D:\CS\test\src\Test2.java:5:72: Abbreviation in name 'ppPOINT' must contain no more than '4' consecutive capital letters. [AbbreviationAsWordInName]
Audit done.
Checkstyle ends with 2 errors.
PS D:\CS\test> 
```

Describe what you expect in detail.

add input file for record pattern syntax

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/naming/AbbreviationAsWordInNameCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/naming/AbbreviationAsWordInNameCheckTest.java
@@ -454,6 +454,28 @@ public class AbbreviationAsWordInNameCheckTest extends AbstractModuleTestSupport
                 expected);
     }
 
+    @Test
+    public void testAbbreviationAsWordInNameCheckRecordPatterns()
+            throws Exception {
+
+        final int expectedCapitalCount = 4;
+
+        final String[] expected = {
+            "23:39: " + getWarningMessage("POINT", expectedCapitalCount),
+            "27:60: " + getWarningMessage("COLOR", expectedCapitalCount),
+            "31:53: " + getWarningMessage("INTEGER", expectedCapitalCount),
+            "31:71: " + getWarningMessage("STRING", expectedCapitalCount),
+            "39:52: " + getWarningMessage("COLOR", expectedCapitalCount),
+            "40:52: " + getWarningMessage("INTEGER", expectedCapitalCount),
+            "40:68: " + getWarningMessage("COLOR", expectedCapitalCount),
+        };
+
+        verifyWithInlineConfigParser(
+                getNonCompilablePath(
+                        "InputAbbreviationAsWordInNameCheckRecordPatterns.java"),
+                expected);
+    }
+
     private String getWarningMessage(String typeName, int expectedCapitalCount) {
         return getCheckMessage(MSG_KEY, typeName, expectedCapitalCount);
     }

--- a/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/InputAbbreviationAsWordInNameCheckRecordPatterns.java
+++ b/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/InputAbbreviationAsWordInNameCheckRecordPatterns.java
@@ -1,0 +1,51 @@
+/*
+AbbreviationAsWordInName
+allowedAbbreviationLength = (default)3
+allowedAbbreviations = (default)
+ignoreFinal = (default)true
+ignoreStatic = (default)true
+ignoreStaticFinal = (default)true
+ignoreOverriddenMethods = (default)true
+tokens = (default)CLASS_DEF, INTERFACE_DEF, ENUM_DEF, ANNOTATION_DEF, ANNOTATION_FIELD_DEF, \
+         PARAMETER_DEF, VARIABLE_DEF, METHOD_DEF, PATTERN_VARIABLE_DEF, RECORD_DEF, \
+         RECORD_COMPONENT_DEF
+
+
+*/
+
+//non-compiled with javac: Compilable with Java21
+package com.puppycrawl.tools.checkstyle.checks.naming.abbreviationaswordinname;
+
+public class InputAbbreviationAsWordInNameCheckRecordPatterns {
+
+    void m(Object o) {
+        // violation below, 'Abbreviation in name 'POINT'*.'
+        if (o instanceof ColoredPoint POINT) {
+
+        }
+        // violation below,  'Abbreviation in name 'COLOR'*.'
+        if (o instanceof ColoredPoint(int x, int y, String COLOR)) {
+
+        }
+
+        if (o instanceof Rectangle(ColoredPoint(int INTEGER,_, String STRING),ColoredPoint _)) {}
+        // 2 violations above:
+        //                'Abbreviation in name 'INTEGER'*.'
+        //               'Abbreviation in name 'STRING'*.'
+    }
+    void m2(Object o) {
+        switch (o) {
+            // violation below, 'Abbreviation in name 'COLOR'*.'
+            case ColoredPoint(int x, int y, String COLOR)  -> {}
+            case Rectangle(ColoredPoint(int x, int INTEGER, String COLOR), _)  -> {}
+            // 2 violations above:
+            //                  'Abbreviation in name 'INTEGER'*.'
+            //                 'Abbreviation in name 'COLOR'*.'
+            default -> {}
+        }
+    }
+
+
+    record ColoredPoint(int x, int y, String color) {}
+    record Rectangle(ColoredPoint upperLeft, ColoredPoint lowerRight) {}
+}

